### PR TITLE
feat(0.1.2): align with API version 1.0.17

### LINKED ISSUE #30: Verify `ignoreScm` option in the testsuite
See https://github.com/fossology/fossology/issues/1755

### PR DESCRIPTION
**Pull request by deveaud-m**
* Add `groupName` option wherever available, handle 403 error code
* Fix `ignore_scm` option for file upload, default shall be false
* Add test to verify `.git` repository has been added to upload correctly
* Add support for the extended `/uploads` endpoint using additional options
* Extend testsuite to verify uploads filtering
* Use the request option `params` wherever possible
* Some minor cosmetic changes in the docstrings

Closes #30

### CODE REVIEW

File: fossology/uploads.py [hunk 1]
```diff
@@ -216,19 +215,16 @@ def upload_licenses(self, upload, agent=None, containers=False):
         :rtype: list of licenses as JSON object
         :raises FossologyApiError: if the REST call failed
         """
+        params = {}
         if agent:
-            agent = agent.value
+            params["agent"] = agent.value
```
Comment (GMishx) [inline fossology/uploads.py:220]: Just a suggestion here, you can actually ask for multiple agents at once. Just pass them as a csv (nomos,monk,ojo).

Comment (deveaud-m) [reply fossology/uploads.py]: :+1: code is aligned now.

**Comment by deveaud-m** [comment]
Coverage check is somehow broken, next PR will address tests and improve coverage.

I'll merge this PR and now and continue the integration with next big API change.

Hope that's ok for you @GMishx

**Comment by GMishx** [comment]
That is certainly ok and sorry I couldn't find time to review your PR.

**Pull request merged**